src/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -1183,7 +1183,7 @@ void CodeGen::genSSE41Intrinsic(GenTreeHWIntrinsic* node)
                 if (baseType == TYP_FLOAT)
                 {
                     // extract instructions return to GP-registers, so it needs int size as the emitsize
-                    emit->emitIns_SIMD_R_R_I(ins, emitTypeSize(TYP_INT), op1Reg, tmpTargetReg, (int)i);
+                    emit->emitIns_SIMD_R_R_I(ins, emitTypeSize(TYP_INT), tmpTargetReg, op1Reg, (int)i);
```

Comment:
I swapped the operands of floating extract because `extractps` uses `RMI` but encodes the target in `r/m`.
![image](https://user-images.githubusercontent.com/1263030/37563577-31b7b03e-2a41-11e8-8149-765f5f576366.png)

Reply:
I don't think that those letters are meant to describe 'how' it is encoded:
![image](https://user-images.githubusercontent.com/10487869/37566683-dd2b636a-2a79-11e8-922b-f48c72745c82.png)

The entry explicitly calls out that `Operand 1` is encoded in the ModRM byte in the `r/m` field and that `Operand 2` is encoded in the ModRM byte in the `reg` field. Since `Operand 1` is encoded in the `r/m` field, this makes it an `MRI` instruction.

Reply:
That is, the only difference between MR and RM is whether Op1 or Op2 is encoded in the `r/m` field of the ModRM byte. MR encodes `Op1` there and RM encodes `Op2` there.